CODE/Sources/PennyBOT/ProposalsChecker.swift

```diff
@@ -0,0 +1,191 @@
+import Logging
+import DiscordModels
+import PennyModels
+import Foundation
+
+actor ProposalsChecker {
+    var previousProposals = [Proposal]()
+    var isFirstRound = true
+
+    let logger = Logger(label: "ProposalsChecker")
+
+    var proposalsService: (any ProposalsService)!
+    var discordService: DiscordService {
+        DiscordService.shared
+    }
+
+    static let shared = ProposalsChecker()
+
+    private init() { }
+
+    func initialize(proposalsService: any ProposalsService) {
+        self.proposalsService = proposalsService
+    }
+
+    nonisolated func run() {
+        Task {
+            do {
+                try await self.check()
+                try await Task.sleep(for: .seconds(60 * 15)) /// 15 mins
+            } catch {
+                logger.report("Couldn't check proposals", error: error)
+            }
+            self.run()
+        }
+    }
+
+    func check() async throws {
+        let proposals = try await self.proposalsService.get()
+
+        /// If it's the first round, we have nothing to compare to
+        if isFirstRound {
+            self.previousProposals = proposals
+            isFirstRound = false
+            return
+        }
```

Comment:
```suggestion
        guard !self.previousProposals.isEmpty else {
            self.previousProposals = proposals
            return
        }
```

Reply:
Thanks, i'll add this change with the next cleanup commit that's incoming

Reply:
Though i have something against `guard !thing else { }`, so i'll just use `if thing { }` 😅